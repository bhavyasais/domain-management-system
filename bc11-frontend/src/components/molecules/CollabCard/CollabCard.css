.MuiCard-root {
    overflow: hidden;
    max-width: 194px;
}
