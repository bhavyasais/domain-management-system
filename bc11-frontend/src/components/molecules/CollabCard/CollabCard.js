import React, { useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardActions from "@material-ui/core/CardActions";
import CardMedia from "@material-ui/core/CardMedia";
import ButtonAtom from "../../atoms/Button/ButtonAtom";
import Typography from "@material-ui/core/Typography";
import "./CollabCard.css";
import icon from "../../../assets/atoms_icon_excel.svg";

const useStyles = makeStyles({
  root: {
    maxWidth: "194px",
    height: "200px",
  },
  media: {
    height: "60px",
    width: "62px",
    padding: "35px 66px 92px 66px",
  },
  media1: {
    height: "18px",
    width: "20px",
    transform: "translate(165px, 8px)",
  },
  action: {
    objectFit: "inherit",
    backgroundColor: "#e8ecef",
    borderRadius: "4px",
    height: "148px",
  },
  button: {
    height: "52px",
  },
  addb: {
    width: "78px",
    height: "32px",
    backgroundColor: "#1d4cd7",
    color: "#ffffff",
    margin: "-8px 10px 8px 4px",
  },
  preview: {
    width: "86px",
    height: "32px",
    backgroundColor: "#ffffff",
    color: "#949494",
    margin: "-8px 2px 8px -10px",
  },
  CardHoverContent: {
    color: "#ffffff",
    position: "absolute",
    bottom: "0px",
    left: "12px",
    fontSize: "18px",
  },
  CardContent: {
    color: "#1c1c21",
    fontSize: "16px",
    bottom: "16px",
    transform: "translate(5%, 35%)",
  },
});

export default function CollabCard(props) {
  const classes = useStyles();
  const [showContent, setShowContent] = useState(true);
  const [showCardContent, setShowCardContent] = useState(false);

  const handleMouseEnter = () => {
    setShowContent(false);
    setShowCardContent(true);
  };
  const handleMouseLeave = () => {
    setShowContent(true);
    setShowCardContent(false);
  };
  const handleOnClick = () => {
    props.onClick();
  };
  const style = showCardContent
    ? { backgroundColor: "#666666", opacity: 0.8, borderRadius: "4px" }
    : {};
  const media_style = showCardContent ? { opacity: 0.2 } : {};
  return (
    <Card
      className={classes.root}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      children={props.children}
    >
      <CardActionArea className={classes.action} style={style}>
        {showCardContent && (
          <Typography
            gutterBottom
            variant="h5"
            color="textSecondary"
            component="h2"
            size="16px"
            className={classes.CardHoverContent}
          >
            {props.name}
          </Typography>
        )}
        <CardMedia
          style={media_style}
          className={classes.media1}
          component={props.component}
          image={icon}
          title={props.title}
        />

        <CardMedia
          style={media_style}
          className={classes.media}
          component={props.component}
          image={props.image}
          title={props.title}
        />
      </CardActionArea>

      {showContent && (
        <CardActions>
          <Typography
            gutterBottom
            variant="h5"
            color="textSecondary"
            component="h2"
            size="16px"
            className={classes.CardContent}
          >
            {props.name}
          </Typography>
        </CardActions>
      )}

      {showCardContent && (
        <CardActions className={classes.button}>
          <ButtonAtom
            className={classes.addb}
            variant="contained"
            value="Add"
            textVariant="caption"
            onClick={handleOnClick}
          />

          <ButtonAtom
            className={classes.preview}
            variant="outlined"
            value="Preview"
            textVariant="caption"
          />
        </CardActions>
      )}
    </Card>
  );
}
