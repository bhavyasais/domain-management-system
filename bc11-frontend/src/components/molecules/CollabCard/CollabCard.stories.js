import React from "react";
import CollabCard from "./CollabCard"; // This is our component
import { storiesOf } from "@storybook/react";
import Card from "@material-ui/core/Card";
import {select} from '@storybook/addon-knobs';
import box from "../../../assets/atoms_icons_box.svg";
import gsuite from "../../../assets/g-suite.jpg";
import okta from "../../../assets/atoms_icons_okta.svg";

const options = {
  box: (
    <CollabCard
      title="box card"
      image={box}
      component="img"
      height="200"
      name="Box"
    />
  ),
  gsuite: (
    <CollabCard
      title="g suite card"
      image={gsuite}
      component="img"
      height="200"
      name="Google Suite"
    />
  ),
  docu: (
    <CollabCard
      title="docu sign card"
      image="https://documentautomation.net/wp-content/uploads/2016/05/docusign-circle-large.png"
      component="img"
      height="200"
      name="Docusign"
    />
  ),
  figma: (
    <CollabCard
      title="figma card"
      image="https://cdn.worldvectorlogo.com/logos/figma-1.svg"
      component="img"
      height="200"
      name="Figma"
    />
  ),
  okta: (
    <CollabCard
      title="okta card"
      image={okta}
      component="img"
      height="200"
      name="Okta"
    />
  ),
  slack: (
    <CollabCard
      title="slack card"
      image="https://cdn.mos.cms.futurecdn.net/SDDw7CnuoUGax6x9mTo7dd.jpg"
      component="img"
      height="200"
      name="Slack"
    />
  ),
};

storiesOf("Molecules/Cards", module).add("All Cards", () => (
  <Card children={options[select("Card", ["box", "gsuite","docu","figma","okta","slack"], options.box)]}  />
));
