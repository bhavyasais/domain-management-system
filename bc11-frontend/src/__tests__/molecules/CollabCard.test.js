import React from 'react';
import { mount, shallow } from 'enzyme';
import CollabCard from '../../components/molecules/CollabCard/CollabCard';

import { act } from 'react-dom/test-utils';
import { Card } from '@material-ui/core';

describe('<CollabCard/>', () => {
  let component;
  const handleClick = jest.fn();
  beforeEach(() => {
    act(() => {
      component = mount(<CollabCard onClick={handleClick} />);
    });
  });

  it('Renders correctly', () => {
    let wrapper;
    wrapper = mount(<CollabCard />);
    expect(wrapper).toMatchSnapshot();
  });

  it('opens the modal', () => {
    const wrapper = mount(<CollabCard onClick={handleClick} />);
    wrapper.find(Card).at(0).props().onMouseEnter();
    wrapper.find(Card).at(0).props().onMouseLeave();
    expect(wrapper).toMatchSnapshot();
  });
});
