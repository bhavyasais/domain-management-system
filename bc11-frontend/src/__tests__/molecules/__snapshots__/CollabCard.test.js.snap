// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollabCard/> Renders correctly 1`] = `ReactWrapper {}`;

exports[`<CollabCard/> opens the modal 1`] = `ReactWrapper {}`;
